### PR TITLE
cargo: Update dependencies identified by audit

### PR DESCRIPTION
**Pull request by jodh-intel**
Ran `cargo audit fix` to update all crates with available security fixes.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

File: ttrpc-codegen/src/parser.rs [hunk 1]
```diff
@@ -2182,7 +2182,7 @@ mod test {
         "#;
 
         let err = FileDescriptor::parse(msg).err().expect("err");
-        assert_eq!(4, err.line);
+        assert_eq!(3, err.line);
```
Comment (jodh-intel) [inline ttrpc-codegen/src/parser.rs:2185]: @Tim-Zhang, @lifupan - I've made this change to get the tests to pass, however, please can you check this is a valid change. Looking at the history, there have been a few fd leak fixes which might explain why this change is needed?

**Comment by jodh-intel** [comment]
Spotted a comment I haven't yet applied so adding dnm...

File: ttrpc-codegen/src/parser.rs (hunk 1 above)
Comment (Tim-Zhang) [reply ttrpc-codegen/src/parser.rs:2185]: It's a valid change.

File: Cargo.toml [hunk 2]
```diff
@@ -14,13 +14,13 @@ description = "A Rust version of ttrpc."
...
 libc = { version = "0.2.59", features = [ "extra_traits" ] }
-nix = "0.16.1"
+nix = "^0.20.2"
 log = "0.4"
 byteorder = "1.3.2"
 thiserror = "1.0"
 
 async-trait = { version = "0.1.31", optional = true }
-tokio = { version = "1", features = ["rt", "sync", "io-util", "macros", "time"], optional = true }
+tokio = { version = ">=1.5.1, <1.6.0", features = ["rt", "sync", "io-util", "macros", "time"], optional = true }
```
Comment (Tim-Zhang) [inline Cargo.toml:23]: Why we need limit the tokio version so strict? and if we need do this, how about using `version = "~1.5.1"`

File: ttrpc-codegen/src/parser.rs (hunk 1 above)
Comment (Tim-Zhang) [reply ttrpc-codegen/src/parser.rs:2185]: The unit test has never passed before, thank you

File: Cargo.toml (hunk 2 above)
Comment (jodh-intel) [reply Cargo.toml]: This was the value produced by `cargo audit fix`, but I agree, your version makes more sense :smile:

**Comment by jodh-intel** [comment]
Thanks @Tim-Zhang - branch updated.

File: Cargo.toml (hunk 2 above)
Comment (Tim-Zhang) [reply Cargo.toml]: But the too strict limit will limit the version selection of our user, if they select the tokio@1.13, and we force the tokio version to `>=1.5.1, <1.6.0`, the error: ` failed to select a version for `tokio` will be met.

Comment (Tim-Zhang) [reply Cargo.toml]: After I have changed the tokio version to 1.13 in examples, I got:
image

**Comment by Tim-Zhang** [comment]
Hi @mxpv, There are some errors with our MacOS BVT, PTAL

File: Cargo.toml (hunk 2 above)
Comment (jodh-intel) [reply Cargo.toml]: Hi @Tim-Zhang - If you change the tokio version in `examples/Cargo.toml`, you also need to do the same to `Cargo.toml` afaik.

ftr, I can run `cargo run --example async-server` successfully using the versions on this PR.

Comment (Tim-Zhang) [reply Cargo.toml]: Think of the examples as a user's program, if the user update the version of tokio in his/her program, how can he/she do the same to the Cargo.toml in ttrpc-rust?

Comment (jodh-intel) [reply Cargo.toml]: Hi @Tim-Zhang - Ah - I see what you're saying now, but I'm not sure how we resolve this.

| branch | main code | examples | Notes |
|-|-|-|-|
| `master` | `1` | `1.0.1` | To permissive? | 
| this PR (take 1) |  `>=1.5.1, <1.6.0` | `>=1.5.1, <1.6.0` | Safe, but too strict? |
| this PR (take 2) | `~1.5.1` | `~1.5.1` | Safer than `master`, but still too strict? |

Do you think this PR should only update the `nix` version then (so that `tokio = "1"` still for the main code, but maybe `tokio = "~1.5.1"` for the example code)? That is the most "flexible" solution for users of this crate I guess. However, it would allow them to use an old (and potentially insecure) version of tokio.

**Comment by mxpv** [comment]
@jodh-intel could you please rebase against latest changes in `master`? I think we need those patches to pass CI on macos.

**Comment by jodh-intel** [comment]
@mxpv - np, done.

File: Cargo.toml [hunk 3]
```diff
@@ -14,13 +14,13 @@ description = "A Rust version of ttrpc."
 protobuf = { version = "2.0", optional = true }
 bytes = { version = "0.4.11", optional = true }
 libc = { version = "0.2.59", features = [ "extra_traits" ] }
-nix = "0.16.1"
+nix = "^0.20.2"
```
Comment (Tim-Zhang) [inline Cargo.toml:17]: The '^' is redundant, cargo treats versions that do not have any operators as they have '^'.
Although it can be deleted or not, but for consistency reasons I think it should be deleted

File: example/Cargo.toml [hunk 4]
```diff
@@ -17,10 +17,10 @@ libc = "0.2.79"
 byteorder = "1.3.2"
 log = "0.4.6"
 simple-logging = "2.0.2"
-nix = "0.16.1"
+nix = "^0.20.2"
```
Comment (Tim-Zhang) [inline example/Cargo.toml:20]: The same reason as above, the '^' should be deleted.

File: Cargo.toml (hunk 2 above)
Comment (Tim-Zhang) [reply Cargo.toml]: @jodh-intel I think the solution that only update the nix version is ok.

Comment (jodh-intel) [reply Cargo.toml]: @Tim-Zhang - ok. Branch updated.

File: Cargo.toml (hunk 3 above)
Comment (jodh-intel) [reply Cargo.toml]: Fixed.

File: example/Cargo.toml (hunk 4 above)
Comment (jodh-intel) [reply example/Cargo.toml]: Fixed.

**Review by Tim-Zhang — APPROVED** [review]
@jodh-intel LGTM, thanks

**Comment by jodh-intel** [comment]
@mxpv - sorry, I forgot to fetch your commit before re-pushing. Could you add back that `pipe2` fix please?

**Comment by mxpv** [comment]
@jodh-intel done

**Review by mxpv — APPROVED** [review]
